src/cfnlint/data/ExtendedSpecs/all/08_registry_value_types.json

```diff
@@ -0,0 +1,12809 @@
+      "AllowedPattern": "^arn:aws[a-z\\-]*:iam::\\d{12}:role/?[a-zA-Z_0-9+=,.@\\-_/]+$",
+      "StringMax": 2048,
+      "StringMin": 20
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::DeviceFleet.Tag.Key",
+    "value": {
+      "AllowedPattern": "^((?!aws:)[\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$",
+      "StringMax": 128,
+      "StringMin": 1
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::DeviceFleet.Tag.Value",
+    "value": {
+      "AllowedPattern": "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.JobDefinitionArn",
+    "value": {
+      "StringMax": 256,
+      "StringMin": 1
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.JobDefinitionName",
+    "value": {
+      "AllowedPattern": "^[a-zA-Z0-9](-*[a-zA-Z0-9])*$"
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.ModelBiasBaselineConfig.BaseliningJobName",
+    "value": {
+      "AllowedPattern": "^[a-zA-Z0-9](-*[a-zA-Z0-9])*$",
+      "StringMax": 63,
+      "StringMin": 1
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.ConstraintsResource.S3Uri",
+    "value": {
+      "AllowedPattern": "^(https|s3)://([^/]+)/?(.*)$"
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.ModelBiasAppSpecification.ImageUri",
+    "value": {
+      "AllowedPattern": ".*"
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.ModelBiasAppSpecification.ConfigUri",
+    "value": {
+      "AllowedPattern": ".*"
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.EndpointInput.EndpointName",
+    "value": {
+      "AllowedPattern": "^[a-zA-Z0-9](-*[a-zA-Z0-9])*"
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.EndpointInput.LocalPath",
+    "value": {
+      "AllowedPattern": ".*"
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.EndpointInput.S3DataDistributionType",
+    "value": {
+      "AllowedValues": [
+        "FullyReplicated",
+        "ShardedByS3Key"
+      ]
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.EndpointInput.S3InputMode",
+    "value": {
+      "AllowedValues": [
+        "Pipe",
+        "File"
+      ]
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.EndpointInput.StartTimeOffset",
+    "value": {
+      "AllowedPattern": "^.?P.*",
+      "StringMax": 15,
+      "StringMin": 1
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::SageMaker::ModelBiasJobDefinition.EndpointInput.EndTimeOffset",
+    "value": {
+      "Allowed
```

Comment:
Similar to https://github.com/aws-cloudformation/cloudformation-cli/issues/459, we need to think about how to verify regular expressions are valid in Java and Python
(this one doesn't seem to be valid in either though 🤔)

Reply:
[`ERROR: Failed building wheel for regex`](https://travis-ci.com/github/aws-cloudformation/cfn-python-lint/jobs/471402886)